azurelinuxagent/ga/update.py

```diff
@@ -666,13 +667,19 @@ def _report_status(self, exthandlers_handler):
             supports_fast_track = False
         else:
             supports_fast_track = self._goal_state.extensions_goal_state.source_channel == GoalStateChannel.HostGAPlugin
+
         vm_status = exthandlers_handler.report_ext_handlers_status(
             incarnation_changed=self._processing_new_extensions_goal_state(),
             vm_agent_update_status=vm_agent_update_status,
             vm_agent_supports_fast_track=supports_fast_track)
-        if vm_status is None:
-            return
 
+        if vm_status is not None:
+            self._report_extensions_summary(vm_status)
+            if self._goal_state is not None:
+                agent_status = exthandlers_handler.get_ext_handlers_status_debug_info(vm_status)
+                self._goal_state.save_to_history(agent_status, AGENT_STATUS_FILE)
```

Comment:
maybe create an abstraction for this too? i.e. `self._goal_state.save_reported_status()` or something?

Reply:
currently i'm using save_to_history to save the manifests as well, i'm planning on getting rid of it in another round of refactoring